components/core/services/nz-dom-event.service.ts

```diff
@@ -0,0 +1,69 @@
+/**
+ * @license
+ * Copyright Alibaba.com All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/LICENSE
+ */
+
+import { Injectable, NgZone, Renderer2 } from '@angular/core';
+import { Observable, Subject } from 'rxjs';
+import { auditTime } from 'rxjs/operators';
+
+interface Listener {
+  handler(e: Event): void;
+  unsubscribe?(): void;
+  countOfListeners: number;
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class NzDomEventService {
+  private readonly resizeSource = new Subject<void>();
+  private readonly domEventListeners = new Map<string, Listener>();
+
+  constructor(private ngZone: NgZone, private renderer: Renderer2) {}
```

Comment:
constructor(rendererFactory2: RendererFactory2) {
    this.renderer = rendererFactory2.createRenderer(null, null);
  }